src/hydranfc_v2/hydranfc_v2_bbio_card_emulator.c

```diff
@@ -0,0 +1,253 @@
+	 */
+	mode_con1.proto.config.spi.dev_gpio_pull = MODE_CONFIG_DEV_GPIO_NOPULL;
+	//mode_con1.proto.config.spi.dev_speed = 5; /* 5 250 000 Hz */
+	mode_con1.proto.config.spi.dev_speed = 6; /* 10 500 000 Hz */
+	mode_con1.proto.config.spi.dev_phase = 1;
+	mode_con1.proto.config.spi.dev_polarity = 0;
+	mode_con1.proto.config.spi.dev_bit_lsb_msb = DEV_FIRSTBIT_MSB;
+	mode_con1.proto.config.spi.dev_mode = DEV_MASTER;
+	bsp_spi_init(BSP_DEV_SPI2, &mode_con1.proto);
+
+	/*
+	 * Initializes the SPI driver 1. The SPI1 signals are routed as follows:
+	 * Shall be configured as SPI Slave for ST25R3916 NFC data sampling on MOD pin.
+	 * NSS. (Not used use Software).
+	 * ST25R3916 MCU_CLK pin28 output / HydraBus PA5 - SCK.(AF5) => SPI Slave CLK input (Sniffer mode/RX Transparent Mode)
+	 * ST25R3916 ST25R3916 MOSI SPI pin31 (IN) / HydraBus PA6 - MISO.(AF5) (MCU TX Transparent Mode)
+	 * ST25R3916 ST25R3916 MISO_SDA pin32 (OUT) / HydraBus PA7 - MOSI.(AF5) => SPI Slave MOSI input (Sniffer mode/RX Transparent Mode)
+	 */
+	/* spiStart() is done in sniffer see sniffer.c */
+	/* HydraBus SPI1 Slave CLK input */
+	palSetPadMode(GPIOA, 5, PAL_MODE_ALTERNATE(5) | PAL_STM32_OSPEED_MID1);
+	/* HydraBus SPI1 Slave MISO. Not used/Not connected */
+	palSetPadMode(GPIOA, 6, PAL_MODE_ALTERNATE(5) | PAL_STM32_OSPEED_MID1);
+	/* HydraBus SPI1 Slave MOSI. connected to ST25R3916 MOD Pin */
+	palSetPadMode(GPIOA, 7, PAL_MODE_ALTERNATE(5) | PAL_STM32_OSPEED_MID1);
+
+	/* Configure K1/2 Buttons as Input */
+	palSetPadMode(GPIOB, 8, PAL_MODE_INPUT); /* K1 Button */
+	palSetPadMode(GPIOB, 9, PAL_MODE_INPUT); /* K2 Button */
+
+	/* Configure D1/2/3/4 LEDs as Output */
+	D1_OFF;
+	D2_OFF;
+	D3_OFF;
+	D4_OFF;
+
+	palSetPadMode(GPIOB, 0, PAL_MODE_OUTPUT_PUSHPULL | PAL_STM32_OSPEED_MID1);
+
+#ifndef MAKE_DEBUG
+	// can't use LED on PB3 if using SWO
+	palSetPadMode(GPIOB, 3, PAL_MODE_OUTPUT_PUSHPULL | PAL_STM32_OSPEED_MID1);
+#endif
+
+	palSetPadMode(GPIOB, 4, PAL_MODE_OUTPUT_PUSHPULL | PAL_STM32_OSPEED_MID1);
+	palSetPadMode(GPIOB, 5, PAL_MODE_OUTPUT_PUSHPULL | PAL_STM32_OSPEED_MID1);
+
+	palDisablePadEvent(GPIOA, 1);
+	/* ST25R3916 IRQ output / HydraBus PA1 input */
+	palClearPad(GPIOA, 1);
+	palSetPadMode(GPIOA, 1, PAL_MODE_INPUT | PAL_STM32_OSPEED_MID1);
+	/* Activates the PAL driver callback */
+	//palDisablePadEvent(GPIOA, 1);
+	palEnablePadEvent(GPIOA, 1, PAL_EVENT_MODE_RISING_EDGE);
+	palSetPadCallback(GPIOA, 1, &extcb1, NULL);
+
+	/* Init st25r3916 IRQ function callback */
+	st25r3916_irq_fn = st25r3916Isr;
+	hal_st25r3916_spiInit(ST25R391X_SPI_DEVICE);
+	if (hydranfc_v2_init_RFAL(con) != ERR_NONE) {
+		cprintf(con, "HydraNFC v2 not found.\r\n");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static void deinit_gpio_spi_nfc(t_hydra_console *con)
+{
+	(void)(con);
+	palClearPad(GPIOA, 1);
+	palSetPadMode(GPIOA, 1, PAL_MODE_INPUT);
+	palDisablePadEvent(GPIOA, 1);
+
+	bsp_spi_deinit(BSP_DEV_SPI2);
+
+	palSetPadMode(GPIOA, 5, PAL_MODE_INPUT);
```

Comment:
To be refactored using https://github.com/hydrabus/hydrafw_hydranfc_shield_v2/blob/master/src/hydranfc_v2/hydranfc_v2_nfc_mode.c#L219